test/performance/benchmarks/rollout-probe/continuous/main.go

```diff
@@ -0,0 +1,131 @@
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"time"
+
+	vegeta "github.com/tsenart/vegeta/lib"
+
+	"knative.dev/pkg/signals"
+	"knative.dev/pkg/test/mako"
+	"knative.dev/serving/test/performance"
+	"knative.dev/serving/test/performance/metrics"
+)
+
+var (
+	target   = flag.String("target", "", "The target to attack.")
+	duration = flag.Duration("duration", 5*time.Minute, "The duration of the probe")
+)
+
+func main() {
+	flag.Parse()
+
+	// We want this for properly handling Kubernetes container lifecycle events.
+	ctx := signals.NewContext()
+
+	// We cron quite often, so make sure that we don't severely overrun to
+	// limit how noisy a neighbor we can be.
+	ctx, cancel := context.WithTimeout(ctx, *duration+time.Minute)
+	defer cancel()
+
+	// Use the benchmark key created
+	mc, err := mako.Setup(ctx)
+	if err != nil {
+		log.Fatal("Failed to setup mako: ", err)
+	}
+	q, qclose, ctx := mc.Quickstore, mc.ShutDownFunc, mc.Context
+	// Use a fresh context here so that our RPC to terminate the sidecar
+	// isn't subject to our timeout (or we won't shut it down when we time out)
+	defer qclose(context.Background())
+
+	// Wrap fatalf in a helper or our sidecar will live forever.
+	fatalf := func(f string, args ...interface{}) {
+		qclose(context.Background())
+		log.Fatalf(f, args...)
+	}
+
+	// Validate flags after setting up "fatalf" or our sidecar will run forever.
+	if *target == "" {
+		fatalf("Missing flag: -target")
+	}
+
+	// Based on the "target" flag, load up our target benchmark.
+	// We only run one variation per run to avoid the runs being noisy neighbors,
+	// which in early iterations of the benchmark resulted in latency bleeding
+	// across the different workload types.
+	t, ok := targets[*target]
+	if !ok {
+		fatalf("Unrecognized target: %s", *target)
+	}
+
+	// Make sure the target is ready before sending the large amount of requests.
+	if err := performance.ProbeTargetTillReady(t.target.URL, *duration); err != nil {
+		fatalf("Failed to get target ready for attacking: %v", err)
+	}
+
+	// Set up the threshold analyzers for the selected benchmark.  This will
+	// cause Mako/Quickstore to analyze the results we are storing and flag
+	// things that are outside of expected bounds.
+	q.Input.ThresholdInputs = append(q.Input.ThresholdInputs, t.analyzers...)
+
+	// Send 2000 QPS (2 per ms) for the given duration with a 30s request timeout.
+	rate := vegeta.Rate{Freq: 3, Per: time.Millisecond}
```

Comment:
`2` in the comment and `3` here, which one is correct?

Reply:
fixed.